src/BenchmarkManager.py

```diff
@@ -266,18 +266,66 @@ def load_config(self, config: dict, app_modules: list) -> None:
                         "devices"][single_device] = device_wrapper
 
     @staticmethod
-    def _query_for_config(config: dict, prefix: str = "") -> dict:
-        for key, config_answer in config.items():
+    def _query_for_config(param_opts: dict, prefix: str = "") -> dict:
+        config = {}
+        for key, config_answer in param_opts.items():
+            if not config_answer.get("if") is None:
+                #
+                #support parameter descriptions like
+                #"seed": {
+                #    "if": {"key":"graph_type", "in" : ["erdos-renyi"]},
+                #    ...
+                #}
+                #meaning that 'seed' only gets displayed if graph_type has been chosen to be 'erdos-renyi'
+                #This expects that the referenced parameter has been declared before and is declared to be
+                #'exclusive' so that its value is unique.
+                #
+
+                key_in_cond = config_answer.get("if")["key"]
+                dependency = param_opts.get(key_in_cond)
+
+                #check configuration is consistent
+                consistent = False
+                err_msg = None
+                if dependency is None:
+                    err_msg = f"Inconsistent parameter options: condition references unknown parameter: {key_in_cond}"
+                elif not dependency.get('exclusive', False):
+                    err_msg = f"Inconsistent parameter options: condition references non exclusive parameter: {key_in_cond}"
+                else:
+                    consistent = True
+                if not consistent:
+                    raise Exception(f"{prefix} {err_msg}")
+                    
+                if not config[key_in_cond][0] in config_answer.get("if")["in"]:
+                    continue
+                    
+            values = None
```

Comment:
I think this line can be removed as `values` will be set either way in the following if/else clauses.

Reply:
```suggestion
```